tests/compile-fail/modinit-used-in-non-fn.rs

```diff
@@ -0,0 +1,7 @@
+#![feature(proc_macro)]
+extern crate pyo3;
+
+use pyo3::{py, PyResult, Python, PyModule};
+
+#[py::modinit(rust2py)]  //~ ERROR: custom attribute panicked
```

Comment:
Tried to use `//~ ERROR: #[modinit] can only be used with fn block` but did not work.
Are we doing something wrong when emitting errors from proc macro?